airflow/changelog.yaml

```diff
@@ -0,0 +1,26 @@
+"Version: 1.10.3":
+    "Authors:": ['author1', 'author2', 'author3']
+    "New Features:":
+        - feature1
+        - feature2
+        - feature3
+    "Bug Fixes:":
+        - fix1
+        - fix2
+        - fix3
+    "Improvements:":
+        - improvement1
+        - improvement2
+"Version: 1.10.2":
+    "Authors:": ['author1', 'author2', 'author3']
+    "New Features:":
+        - feature1
+        - feature2
+        - feature3
+    "Bug Fixes:":
+        - fix1
+        - fix2
+        - fix3
+    "Improvements:":
+        - improvement1
+        - improvement2
```

Comment:
We need to keep this file empty for now.

Reply:
Change yaml content structure to this:
```yaml
changelog:
  - version: 1.10.3
    authors: ['author1', 'author2', 'author3']
    newFeatures:
      - feature1
      - feature2
      - feature3
    bugFixes:
      - fix1
      - fix2
      - fix3
    improvements:
      - improvement1
      - improvement2
  - version: 1.10.2
    authors: ['author1', 'author2', 'author3']
    newFeatures:
      - feature1
      - feature2
      - feature3
    bugFixes:
      - fix1
      - fix2
      - fix3
    improvements:
      - improvement1
      - improvement2
```

Reply:
`changelog:
    "Version: 1.10.3":
        "Authors:": ['author1', 'author2', 'author3']
        "New Features:":
            - feature1
            - feature2
            - feature3
        "Bug Fixes:":
            - fix1
            - fix2
            - fix3
        "Improvements:":
            - improvement1
            - improvement2
    "Version: 1.10.2":
        "Authors:": ['author1', 'author2', 'author3']
        "New Features:":
            - feature1
            - feature2
            - feature3
        "Bug Fixes:":
            - fix1
            - fix2
            - fix3
        "Improvements:":
            - improvement1
            - improvement2
`

This is the updated structure of the changelog file.

Reply:
``` yaml  
changelog:
    "Version: 1.10.3":
        "Authors:": ['author1', 'author2', 'author3']
        "New Features:":
            - feature1
            - feature2
            - feature3
        "Bug Fixes:":
            - fix1
            - fix2
            - fix3
        "Improvements:":
            - improvement1
            - improvement2
    "Version: 1.10.2":
        "Authors:": ['author1', 'author2', 'author3']
        "New Features:":
            - feature1
            - feature2
            - feature3
        "Bug Fixes:":
            - fix1
            - fix2
            - fix3
        "Improvements:":
            - improvement1
            - improvement2

```

Sorry

Reply:
We want to follow the attribute naming conventions.

Please strictly follow https://github.com/coutureai/RaWorkflowOrchestrator/pull/9#discussion_r434410267

Reply:
```yaml
changelog:
    "version: 1.10.3":
        authors: ['author1', 'author2', 'author3']
        newFeatures:
          - feature1
          - feature2
          - feature3
        bugFixes:
          - fix1
          - fix2
          - fix3
        improvements:
          - improvement1
          - improvement2
    "version: 1.10.2":
        authors: ['author1', 'author2', 'author3']
        newFeatures:
          - feature1
          - feature2
          - feature3
        bugFixes:
          - fix1
          - fix2
          - fix3
        improvements:
          - improvement1
          - improvement2
```

Is this fine? Otherwise I will have to make modifications in the loops.

Reply:
The version is not  the way I want. Make modifications in your loop. We want an list, not a dict in the yaml. Notice how my changelog is an array, while that of yours is a dict like object.

Reply:
In that case however, the version name is being treated as one of the headings.

Reply:
You can easily extract that from list as well.

Reply:
I think because the heading names are getting repeated, it won't work. I tried printing and only the data corresponding to version 1.10.2 was getting printed.

Reply:
I disagree. Try it yourself.
Here is an example:
```sh
>>> import yaml
>>> yaml_str = '''changelog:
...   - version: 1.10.3
...     authors: ['author1', 'author2', 'author3']
...     newFeatures:
...       - feature1
...       - feature2
...       - feature3
...     bugFixes:
...       - fix1
...       - fix2
...       - fix3
...     improvements:
...       - improvement1
...       - improvement2
...   - version: 1.10.2
...     authors: ['author1', 'author2', 'author3']
...     newFeatures:
...       - feature1
...       - feature2
...       - feature3
...     bugFixes:
...       - fix1
...       - fix2
...       - fix3
...     improvements:
...       - improvement1
...       - improvement2'''
>>> yaml_str
"changelog:\n  - version: 1.10.3\n    authors: ['author1', 'author2', 'author3']\n    newFeatures:\n      - 
feature1\n      - feature2\n      - feature3\n    bugFixes:\n      - fix1\n      - fix2\n      - fix3\n    i
mprovements:\n      - improvement1\n      - improvement2\n  - version: 1.10.2\n    authors: ['author1', 'aut
hor2', 'author3']\n    newFeatures:\n      - feature1\n      - feature2\n      - feature3\n    bugFixes:\n  
    - fix1\n      - fix2\n      - fix3\n    improvements:\n      - improvement1\n      - improvement2"
>>> data = yaml.load(yaml_str)
<stdin>:1: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is u
nsafe. Please read https://msg.pyyaml.org/load for full details.
>>> data
{'changelog': [{'version': '1.10.3', 'authors': ['author1', 'author2', 'author3'], 'newFeatures': ['feature1
', 'feature2', 'feature3'], 'bugFixes': ['fix1', 'fix2', 'fix3'], 'improvements': ['improvement1', 'improvem
ent2']}, {'version': '1.10.2', 'authors': ['author1', 'author2', 'author3'], 'newFeatures': ['feature1', 'fe
ature2', 'feature3'], 'bugFixes': ['fix1', 'fix2', 'fix3'], 'improvements': ['improvement1', 'improvement2']
}]}
>>> for obj in data['changelog']:
...     print(obj)
... 
{'version': '1.10.3', 'authors': ['author1', 'author2', 'author3'], 'newFeatures': ['feature1', 'feature2', 
'feature3'], 'bugFixes': ['fix1', 'fix2', 'fix3'], 'improvements': ['improvement1', 'improvement2']}
{'version': '1.10.2', 'authors': ['author1', 'author2', 'author3'], 'newFeatures': ['feature1', 'feature2', 
'feature3'], 'bugFixes': ['fix1', 'fix2', 'fix3'], 'improvements': ['improvement1', 'improvement2']}
```